ios/tunnel/rwcendpoint.go

```diff
@@ -0,0 +1,158 @@
+package tunnel
+
+// I took this code from https://github.com/xjasonlyu/tun2socks which is an amazing project
+// but GPL licensed.
+// Technically this would require me from having go-ios be MIT licensed now.
```

Comment:
go-ios has the MIT license 😉 It would need to be GPL licensed, and that would be a problem.

Reply:
yeah it was late when I wrote it, thanks for the comment!